github/Repository.py

```diff
@@ -4085,6 +4086,27 @@ def update_dependabot_alert(
         )
         return github.DependabotAlert.DependabotAlert(self._requester, headers, data, completed=True)
 
+    def get_custom_properties(self) -> dict[str, None | str | list]:
+        """
+        :calls: `GET /repos/{owner}/{repo}/properties/values <https://docs.github.com/en/rest/repos/custom-properties#get-all-custom-property-values-for-a-repository>`_
+        :rtype: dict[str, None | str | list]
+        """
+        url = f"{self.url}/properties/values"
+        _, data = self._requester.requestJsonAndCheck("GET", url)
+        return {p["property_name"]: p["value"] for p in data}
```

Comment:
It looks like the repo [`/`](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository) and [`/properties/values`](https://docs.github.com/en/rest/repos/custom-properties?apiVersion=2022-11-28#get-all-custom-property-values-for-a-repository) return different schemas - I would prefer sticking to `dict[str, *]` to make it easier to work with. Let me know if you think otherwise :)

Reply:
One issue I think can of is if GitHub decides to expand the dict with more properties - but otherwise this would make the API easier to work with.